packages/interactivity/docs/2-api-reference.md

```diff
@@ -471,6 +472,57 @@ store( "myPlugin", {
 
 The `wp-init` can return a function. If it does, the returned function will run when the element is removed from the DOM.
 
+#### `wp-run` 
+
+It runs the passed callback **during node's render execution**.
+
+You can attach several `wp-run` to the same DOM element by using the syntax `data-wp-run--[unique-id]`. _The unique id doesn't need to be unique globally, it just needs to be different than the other unique ids of the `wp-run` directives of that DOM element._
+
+_Example of `data-wp-run` directive_ 
+
+```html
+<div data-wp-run="callbacks.logCounter">
+  <p>Hi!</>
+</div>
+```
+
+_Example of several `wp-run` directives on the same DOM element_
+
+```html
+<div 
+  data-wp-run--log="callbacks.logCounter" 
+  data-wp-run--custom="callbacks.customHooks"
+>
+  <p>Hi!</>
+</div>
+```
+
+<details>
+  <summary><em>See store used with the directive above</em></summary>
+
+```js
+store( "myPlugin", {
+  callbacks: {
+    logCounter: () => {
+      useInit(() => console.log( `Init at ` + new Date() ))
+      useWatch(() => {
+        const { counter } = getContext(); 
+        console.log("Counter is " + counter + " at " + new Date() );
+      })
+    },
+    customHooks: () => {
+      // You can call your own hooks here.
+      useCustomHook( /* ... */ );
+    }
+  },
+} );
```

Comment:
Let's use a more complex example to make sure people understand when they should use `data-wp-run` instead of `data-wp-init`/`watch`:

```suggestion
import { store, useState, useEffect } from '@wordpress/interactivity';

// Unlike `data-wp-init` and `data-wp-watch`, you can use any hooks inside
// `data-wp-run` callbacks.
const useInView = ( ref ) => {
	const [ inView, setInView ] = useState( false );
	useEffect( () => {
		const observer = new IntersectionObserver( ( [ entry ] ) => {
			setInView( entry.isIntersecting );
		} );
		if ( ref ) observer.observe( ref );
		return () => ref && observer.unobserve( ref );
	}, []);
	return inView;
};

store( 'myPlugin', {
  callbacks: {
    customHooks: () => {
      const { ref } = getElement();
      const isInView = useInView( ref );
			useEffect( () => {
				if ( isInView ) {
					console.log( 'Inside' );
				} else {
					console.log( 'Outside' );
				}
			});
    }
  },
} );
```

Reply:
Done in https://github.com/WordPress/gutenberg/pull/57805/commits/d104a0862c2289cbf0b8f9798aa7163b838dba0c.